types/plotly.js/index.d.ts

```diff
@@ -572,7 +574,7 @@ export interface PlotMarker {
 	showscale: boolean;
 	line: Partial<ScatterMarkerLine>;
 	width: number;
-	colorbar: {
+	colorbar: Partial<{
```

Comment:
Wrapping an inline interface in a partial isn't very clean. I would suggest creating an interface for colorbar to clean this up.

```ts
  colorbar: Partial<ColorBar>;
```

Reply:
@szechyjs Fixed!
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29428/commits/4b20dea2b2591730266523927c4f90800dea169b